File: app_admin/lib/features/doctors/data/dto/doctor_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../../../services/data/dto/medical_service_dto.dart';

part 'doctor_dto.freezed.dart';
part 'doctor_dto.g.dart';

@freezed
class DoctorDto with _$DoctorDto {
  const factory DoctorDto({
    required String id,
    required String userId,
    required String fullName,
    required String specialty,
    required String? description,
    required String? phone,
    required String? avatarUrl,
    required double? rating,
    required int? totalReviews,
    @Default(true) bool isAvailable,
    required double? consultationFee,
    required List<MedicalServiceDto>? services,
  }) = _DoctorDto;

  factory DoctorDto.fromJson(Map<String, dynamic> json) =>
      _$DoctorDtoFromJson(json);
}

File: app_be/user-service/src/main/java/com/medibook/user/dto/DoctorDto.java
package com.medibook.user.dto;

import lombok.*;

import java.util.List;
import java.util.UUID;

/**
 * DTO cho Doctor response
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class DoctorDto {
    private UUID id;
    private UUID userId;
    private String fullName;
    private String specialty;
    private String description;
    private String phone;
    private String avatarUrl;
    private Double rating;
    private Integer totalReviews;
    private Boolean isAvailable;
    private Double consultationFee;
    private List<MedicalServiceDto> services;
}
